### DISCUSSION TITLE
Export annotations with labels or is it possible to download videos via Biigle?

### DISCUSSION THREAD

**Discussion by tillfri** [biigle/core#913] [Q&A]
For a project multiple videos have been annotated by domain experts. We now want to export the annotations in order to obtain images which contain the annotations from the videos. So far I have been able to get the annotations as csv data by requesting a report.
1. Is it possible to also download the videos from biigle directly?
2. When viewing the annotations using largo, cropped images containing the annotated object are shown in a grid view. Is it possible to download these or the frame which is used to produce the cropped preview?

**Comment by mzur** [comment]
Hi

> Is it possible to also download the videos from biigle directly?

If the videos are stored on a remote location, you can download them from there. If the videos were uploaded to BIIGLE, you can download them via the API. [Here](https://github.com/biigle/community-resources/tree/master/biigle) is a Python script helper that makes authentication a little easier. You can then use [this API endpoint](https://biigle.de/doc/api/index.html#api-Videos-ShowVideoFile) to get the file. It will redirect you to a temporary S3 URL of the video.

> When viewing the annotations using largo, cropped images containing the annotated object are shown in a grid view. Is it possible to download these or the frame which is used to produce the cropped preview?

There is also a [script for the largo patches](https://github.com/biigle/community-resources/tree/master/grab-largo-patches). You can't get the whole video frame from BIIGLE.
